code/modules/integrated_electronics/subtypes/weaponized.dm

```diff
@@ -0,0 +1,344 @@
+			to_chat(user, "<span class='warning'>There is already a grenade attached!</span>")
+		else if(user.transferItemToLoc(G,src))
+			user.visible_message("<span class='warning'>\The [user] attaches \a [G] to \the [src]!</span>", "<span class='notice'>You attach \the [G] to \the [src].</span>")
+			attach_grenade(G)
+			G.forceMove(src)
+	else
+		return ..()
+
+/obj/item/integrated_circuit/weaponized/grenade/attack_self(var/mob/user)
+	if(attached_grenade)
+		user.visible_message("<span class='warning'>\The [user] removes \an [attached_grenade] from \the [src]!</span>", "<span class='notice'>You remove \the [attached_grenade] from \the [src].</span>")
+		user.put_in_hands(attached_grenade)
+		detach_grenade()
+	else
+		return ..()
+
+/obj/item/integrated_circuit/weaponized/grenade/do_work()
+	if(attached_grenade && !attached_grenade.active)
+		var/datum/integrated_io/detonation_time = inputs[1]
+		var/dt
+		if(isnum(detonation_time.data) && detonation_time.data > 0)
+			dt = CLAMP(detonation_time.data, 1, 12)*10
+		else
+			dt = 15
+		addtimer(CALLBACK(attached_grenade, /obj/item/grenade.proc/prime), dt)
+		var/atom/holder = loc
+		message_admins("activated a grenade assembly. Last touches: Assembly: [holder.fingerprintslast] Circuit: [fingerprintslast] Grenade: [attached_grenade.fingerprintslast]")
+
+// These procs do not relocate the grenade, that's the callers responsibility
+/obj/item/integrated_circuit/weaponized/grenade/proc/attach_grenade(var/obj/item/grenade/G)
+	attached_grenade = G
+	G.forceMove(src)
+	desc += " \An [attached_grenade] is attached to it!"
+
+/obj/item/integrated_circuit/weaponized/grenade/proc/detach_grenade()
+	if(!attached_grenade)
+		return
+	attached_grenade.forceMove(drop_location())
+	attached_grenade = null
+	desc = initial(desc)
+
+/obj/item/integrated_circuit/weaponized/air_cannon
+	name = "pneumatic cannon"
+	desc = "A compact pneumatic cannon to throw things from inside or nearby tiles at a high enough velocity to cause damage. Requires air from a canister to fire."
+	extended_desc = "The first and second inputs need to be numbers which correspond to the coordinates to throw objects at relative to the machine itself. \
+	The 'fire' activator will cause the mechanism to attempt to launch objects at the coordinates, if possible. Note that the \
+	projectile needs to be inside the machine, or on an adjacent tile, and must be medium sized or smaller. The assembly \
+	must also be a gun if you wish to launch something while the assembly is in hand."
+	complexity = 75
+	w_class = WEIGHT_CLASS_SMALL
+	size = 4
+	cooldown_per_use = 30
+	ext_cooldown = 15
+	inputs = list(
+		"target X rel" = IC_PINTYPE_NUMBER,
+		"target Y rel" = IC_PINTYPE_NUMBER,
+		"projectile" = IC_PINTYPE_REF,
+		"canister" = IC_PINTYPE_REF
+		)
+	outputs = list()
+	activators = list(
+		"fire" = IC_PINTYPE_PULSE_IN
+	)
+	spawn_flags = IC_SPAWN_RESEARCH
+	action_flags = IC_ACTION_COMBAT
+	power_draw_per_use = 50
+	var/mole_requirement = 1
```

Comment:
Why is this more efficient than the adminspawn-only pneumatic cannon?

Reply:
Fixed